### ISSUE TITLE
Changed menu structure

### ISSUE THREAD

**Issue by christianalfoni** [christianalfoni/react-webpack-cookbook#4]
Hi there,

You said that change of menu structure would probably break gitbook. I suggested a change now as it felt a bit weird to have those headers as links. There was not really anything to put in there :-)

It would be great, if you agree, to spread some fairy dust on gitbook and make this structure work.

Adding content whenever I get time. Things are starting to come a bit more together now. Looking forward to spreading it like crazy, I think this will be extremely valuable to developers :-)

**Comment by bebraw** [comment]
I can make the structure work. There is one primary question, though. Gitbook works using section/subsection kind of structure. In the structure you are proposing we get subsections. The question is, what to put on the section pages?

One way to deal with it would be to set up a convention. Let's say each chapter (Webpack, React JS, ...) has a certain, named section at the beginning. We could call that an introduction. I would then use the content of that to construct the primary page of the section. After all, that's what its purpose is. It is about introducing the section to the reader.

To summarize you would get a transformation like this:

```markdown
## React JS -> Becomes a section named "React JS"

* Introduction -> Section content (I'll pick the first one or we can use a convention)
* Configuring React JS -> Subsection (same for the rest within this list) 
* ...
```

**Comment by christianalfoni** [comment]
Aha, I see, if we have to have an introduction I think we can just go back
to the previous structure and use a "head introduction" convention. Not
sure what to put there though... maybe summarize the important parts of the
sub section?

Okay, I suggest we just switch back to previous structure and create
"introductions" for each section that summarizes the important points of
the sub sections :-)

**Comment by bebraw** [comment]
> maybe summarize the important parts of the sub section?

It's the perfect place for some background information. So instead of having to start with technical stuff, it is possible to understand why this or that thing is important. It's mostly about providing the reader sufficient motivation to actually go through the material. I can help you to write these and provide initial stubs with some core ideas.

I'll change the structure back to previous and add little introductions for you to look at.

**Comment by christianalfoni** [comment]
Fantastic, thanks :-)

**Comment by bebraw** [comment]
As promised, I expanded the introductions: [React](https://github.com/christianalfoni/react-webpack-cookbook/wiki/Introduction-to-React-JS), [Webpack](https://github.com/christianalfoni/react-webpack-cookbook/wiki/Introduction-to-Webpack).

I'm not sure what that CSS section is about. Can you expand on that?

How about `Advanced`? What would be the gist of that section?

**Comment by christianalfoni** [comment]
Great!

Jup, will expand on CSS. Was thinking of explaining the css loader, style loader, less and imports in less. Fonts, image inlining etc. :-)

In advanced I was thinking of optimizing in general. React hot loader. `Advanced` might not be the right header... its like, not advanced, but more, "wing it like a pro" :-)

**Comment by bebraw** [comment]
> Jup, will expand on CSS. Was thinking of explaining the css loader, style loader, less and imports in less. Fonts, image inlining etc. :-)

Ok. Do you want to do an intro for that or should I write something? It would be a good idea to discuss the history and previous solutions a little bit like I did in mine to provide some context. The subsections can then build on this.

I like the idea of `Wing It Like A Pro`. Now the intent is more clear.

**Comment by christianalfoni** [comment]
Hi, yeah, I saw your introductions. They are great! :-) I am having a hard
time making those introductions, so it would be great if you felt inspired
to write up one on CSS, fonts and images. I will continue to crunch in
details in the meantime.

**Comment by bebraw** [comment]
@christianalfoni I expanded on that CSS, fonts and images introduction. I'm still a bit unsure on what to put to fonts/images parts but those parts will make more sense after I see what you write about them.

**Comment by christianalfoni** [comment]
Awesome :-) Jup, I will continue with details on that stuff as soon as I get a chance

**Comment by bebraw** [comment]
Exhausted this discussion. Better open separate issues as needed.
